src/runtime_src/xrt/device/device.h

```diff
@@ -786,6 +786,24 @@ class device : public xrt_device
     return m_hal->readTrace(type,vec);
   }
 
+  hal::operations_result<void>
```

Comment:
I am not sure I fancy profiling relying on xrt::device thereby adding more methods.
Can profiling stay at the shim level?  I don't see anything in these methods that wrap automatic objects, so all that's added in hal/device.h, hal.h, hal2.h, hal2.cpp are pass through functions to shim.  

 I can imagine that the device handle may be missing thus making it difficult to call shim, but then expose that, rather than adding a slew of new functions and additional profiling specific code that really don't provide any object abstractions.